pkg/scheduler/core/generic_scheduler.go

```diff
@@ -129,6 +130,11 @@ func (g *genericScheduler) Schedule(pod *v1.Pod, nodeLister algorithm.NodeLister
 		return "", ErrNoNodesAvailable
 	}
 
+	// Snapshot all predicate generations and references of node caches.
```

Comment:
It is important that this happens before we snapshot the regular cache. (`UpdateNodeNameToInfoMap` is not the best name for that...) Could you expand on this comment so that future readers know the order matters? (Maybe borrow wording from the deleted comments in factory.go.)

In case it is not clear, the reason we must have it this way is the following scenario.
1. snapshot cache
2. event arrives, updating cache and invalidating ecache
3. snapshot ecache
4. evaluate predicates
5. result is written to ecache... using stale info from the cache snapshot

Reply:
Updated but not sure if it's good enough. Let me know if you have some suggestions about wording.